WPMedia/fusion-news-theme-blocks
# Tmedia 492 stripe integrate

## Description
Building upon the Create New Order work (TMEDIA-517) - update the placeholder area to contain the stripe integration components

## Jira Ticket
- [TMEDIA-492](https://arcpublishing.atlassian.net/browse/TMEDIA-492)

## Acceptance Criteria
As a user who was paywalled I would like to be able to subscribe to the website in order to carry on reading articles without further paywalls

## Test Steps

1. Checkout this branch `git checkout TMEDIA-492-stripe-integrate`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/subscription-block`
3. Go to the checkout page. This should have the checkout block. 
4. Delete the checkout block
5. Re-add the checkout block

<img width="273" alt="Screen Shot 2021-12-17 at 10 37 29" src="https://user-images.githubusercontent.com/5950956/146606028-9bc1a8d4-9fa9-4782-b840-b0524a5c5fe1.png">

6. See that you can set success url and it defaults to '/', Since this is not in client's hands, I'm assuming we can re-add the block for our tests 
7. Sign in to a user
8. Go to the offer page http://localhost/offer/
9. Pick an offer
10. Update first and last name and country in the contact page. Click the next button
11. Input fake credit card details from stripe testing page. 4242424242424242 for number and any future date, cvv, and zip

<img width="883" alt="Screen Shot 2021-12-17 at 14 40 45" src="https://user-images.githubusercontent.com/5950956/146605941-18293509-d983-496b-a97b-e0fe905a2744.png">

12. Click submit 
13. You'll be redirected to the '/' page 
14. Note the order and your subs selection in the subs dashboard 
<img width="1440" alt="Screen Shot 2021-12-16 at 16 30 20" src="https://user-images.githubusercontent.com/5950956/146606066-4800d498-e161-44dc-9a03-7f91c34310af.png">

## Effect Of Changes
### Before

- Showed payment form text

### After

- can submit to stripe payments and the api 

## Dependencies or Side Effects
- the input shown in the designs doesn't match the one subs used. I looked into customizing it as much as I could

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added. -> added code snippets

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
